### PR TITLE
MSP430 fixes

### PR DESCRIPTION
**Pull request by keith-packard**
This series fixes several library bugs and adjusts the test suite to work on 16-bit int targets, and includes scripts to run tests on MSP430 using the binutils-gdb simulator.

### CODE REVIEW

**Comment by keith-packard** [comment]
This series includes two kludge-arounds for bugs in either the compiler or the simulator (but as these both occur on both mspsim and the binutils simulator, I'm suspecting the compiler). I'm happy to leave work-arounds in the library (working is better than broken), but it would be good to root-cause both of them. Both relate to dealing with 20-bit values on targets with support for them, so I wouldn't be surprised if they represented the same underlying bug.

File: newlib/libm/test/math.c [hunk 1]
```diff
@@ -676,8 +678,12 @@ test_math (int vector)
   test_asinh(vector);
   test_asinhf(vector);
   test_atan(vector);
+#endif
+#if TEST_PART > 0
```
Comment (pjonsson) [inline newlib/libm/test/math.c:682]: I think this is missing `|| TEST_PART == -1`.

File: newlib/libm/test/conv_vec.c [hunk 2]
```diff
@@ -3692,5 +3713,7 @@ double_type doubles[] =
 {__LINE__, "0x1.0p-969", 0x1.0p-969, 10 },
 {__LINE__, "0x1.0p-970", 0x1.0p-970, 10 },
 #endif
+#else
```
Comment (pjonsson) [inline newlib/libm/test/conv_vec.c:3716]: Don't think these elses at the end of the arrays are required with the new structure.

Comment (keith-packard) [reply newlib/libm/test/conv_vec.c]: thanks, I'd almost forgotten to fix that. Done.

**Review by pjonsson — APPROVED** [review]
Not sure if you're out of politeness is ignoring my comment about atan2-tests being disabled, but I'll trust your judgement.

The C stuff looks reasonable and the tests are passing. Can't judge the linker stuff on detail level, but it apparently works well enough to run the tests.

File: newlib/libm/test/math.c (hunk 1 above)
Comment (keith-packard) [reply newlib/libm/test/math.c:682]: oops, missed this -- thanks.

**Comment by keith-packard** [comment]
> Not sure if you're out of politeness is ignoring my comment about atan2-tests being disabled, but I'll trust your judgement.

Heh. Nope, I just missed this. Thanks for the reminder.

> The C stuff looks reasonable and the tests are passing. Can't judge the linker stuff on detail level, but it apparently works well enough to run the tests.

**Pull request merged**